test/cmdlineTests.sh

```diff
@@ -112,20 +112,34 @@ function test_solc_behaviour()
         sed -i.bak -E -e 's/ Consider adding \\"pragma solidity \^[0-9.]*;\\"//g' "$stdout_path"
         sed -i.bak -e 's/"errors":\[\],\{0,1\}//' "$stdout_path"
-        # Remove explicit bytecode and references to bytecode offsets
-        sed -i.bak -E -e 's/\"object\":\"[a-f0-9]+\"/\"object\":\"bytecode removed\"/g' "$stdout_path"
-        sed -i.bak -E -e 's/\"opcodes\":\"[^"]+\"/\"opcodes\":\"opcodes removed\"/g' "$stdout_path"
-        sed -i.bak -E -e 's/\"sourceMap\":\"[0-9:;-]+\"/\"sourceMap\":\"sourceMap removed\"/g' "$stdout_path"
+        sed -i.bak -E -e 's/\"opcodes\":\"[^"]+\"/\"opcodes\":\"<OPCODES REMOVED>\"/g' "$stdout_path"
+        sed -i.bak -E -e 's/\"sourceMap\":\"[0-9:;-]+\"/\"sourceMap\":\"<SOURCEMAP REMOVED>\"/g' "$stdout_path"
+
+        # Remove bytecode (but not linker references).
+        sed -i.bak -E -e 's/(\"object\":\")[0-9a-f]+([^"]*\")/\1<BYTECODE REMOVED>\2/g' "$stdout_path"
+        sed -i.bak -E -e 's/(\"object\":\"[^"]+\$__)[0-9a-f]+(\")/\1<BYTECODE REMOVED>\2/g' "$stdout_path"
+        sed -i.bak -E -e 's/(__\$[0-9a-f]{34}\$__)[0-9a-f]+(__\$[0-9a-f]{34}\$__)/\1<BYTECODE REMOVED>\2/g' "$stdout_path"
+
         # Replace escaped newlines by actual newlines for readability
         sed -i.bak -E -e 's/\\n/\'$'\n/g' "$stdout_path"
         rm "$stdout_path.bak"
     else
         sed -i.bak -e '/^Warning: This is a pre-release compiler version, please do not use it in production./d' "$stderr_path"
         sed -i.bak -e '/^Warning (3805): This is a pre-release compiler version, please do not use it in production./d' "$stderr_path"
-        sed -i.bak -e 's/\(^[ ]*auxdata: \)0x[0-9a-f]*$/\1AUXDATA REMOVED/' "$stdout_path"
+        sed -i.bak -e 's/\(^[ ]*auxdata: \)0x[0-9a-f]*$/\1<AUXDATA REMOVED>/' "$stdout_path"
         sed -i.bak -e 's/ Consider adding "pragma .*$//' "$stderr_path"
-        sed -i.bak -e 's/\(Unimplemented feature error: .* in \).*$/\1FILENAME REMOVED/' "$stderr_path"
-        sed -i.bak -e 's/"version": "[^"]*"/"version": "VERSION REMOVED"/' "$stdout_path"
+        sed -i.bak -e 's/\(Unimplemented feature error: .* in \).*$/\1<FILENAME REMOVED>/' "$stderr_path"
+        sed -i.bak -e 's/"version": "[^"]*"/"version": "<VERSION REMOVED>"/' "$stdout_path"
+
+        # Remove bytecode (but not linker references). Since non-JSON output is unstructured,
+        # use metadata markers for detection to have some confidence that it's actually bytecode
+        # and not some random word.
+        # 64697066735822 = hex encoding of 0x64 'i' 'p' 'f' 's' 0x58 0x22
+        # 64736f6c63     = hex encoding of 0x64 's' 'o' 'l' 'c'
+        sed -i.bak -e 's/[0-9a-f]*64697066735822[0-9a-f]\+64736f6c63[0-9a-f]\+/<BYTECODE REMOVED>/g' "$stdout_path"
+        sed -i.bak -e 's/\(__\$[0-9a-f]\{34\}\$__\)[0-9a-f]\+\(__\$[0-9a-f]\{34\}\$__\)/\1<BYTECODE REMOVED>\2/g' "$stdout_path"
+        sed -i.bak -e 's/[0-9a-f]\+\(\(__\$[0-9a-f]\{34\}\$__\)*<BYTECODE REMOVED>\)/<BYTECODE REMOVED>\1/g' "$stdout_path"
```

Comment:
Here again, the detection is less strict than it could be. The first line will only match solidity bytecode (i.e. with proper metadata) but the second will also find bytecode from assembly compilation, which is fine but was not intended. The side-effect is that in case of bytecode from assembly compilation the bytecode at the end is not replaced and at the beginning only is when there are at least two unresolved references. Examples:

```
73__$f919ba91ac99f96129544b80b9516b27a8$__50
```
```
<BYTECODE REMOVED>__$05b0326038374a21e0895480a58bda0768$__<BYTECODE REMOVED>__$fb58009a6b1ecea3b9d99bedd645df4ec3$__5050
```

I could make the regexes more complex (and more lax) to avoid this but this is just a cosmetic issue. Such bytecode does not contain metadata with embedded compiler version or with numbers depending on the size of the version string so it's safe.